client/rpc/src/lib.rs

```diff
@@ -503,18 +528,36 @@ where
         // Construct key challenges and generate key proofs for them.
         let mut key_proofs = KeyProofs::new();
         for file_key in &proven_keys {
-            // Generate the key proof for each file key.
-            let key_proof = generate_key_proof(
-                self.client.clone(),
-                self.file_storage.clone(),
-                *file_key,
-                seed,
-                provider_id,
-                None,
-            )
-            .await?;
-
-            key_proofs.insert(*file_key, key_proof);
+            // If the file key is a checkpoint challenge for a file deletion, we should NOT generate a key proof for it.
+            let should_generate_key_proof = if let Some(checkpoint_challenges) =
+                checkpoint_challenges.clone()
```

Comment:
Same here, using `.as_ref()` would avoid having to clone.

Reply:
Thanks for pointing it out! Done.